src/unix/mod.rs

```diff
@@ -1423,12 +1424,20 @@ cfg_if! {
     if #[cfg(not(any(target_os = "emscripten",
                      target_os = "android",
                      target_os = "haiku",
-                     target_os = "nto")))] {
+                     target_os = "nto",
+                     target_os = "solaris")))] {
         extern "C" {
             pub fn adjtime(delta: *const timeval, olddelta: *mut timeval) -> ::c_int;
         }
     }
 }
+cfg_if! {
+    if #[cfg(target_os = "solaris")] {
```

Comment:
```suggestion
    } else if #[cfg(target_os = "solaris")] {
```

I think this could be combined with the existing `cfg_if`